File: 백준/Gold/1806. 부분합/부분합.cc
#include <iostream>

using namespace std;

int n, s, sum, len = 1e9;
int arr[100000];

int main() {
	ios_base::sync_with_stdio(0);
	cin.tie(0);
	cin >> n >> s;

	for (int i = 0; i < n; i++)
		cin >> arr[i];

	int st = 0;
	int e = 0;
	int cnt = 0;
	while (e <= n) {
		if (sum < s) {
			sum += arr[e++];
			cnt++;
		} else if (sum >= s) {
			len = min(len, cnt);
			sum -= arr[st++];
			cnt--;
		}
	}

	if (len == 1e9)
		cout << 0;
	else
		cout << len;

	return 0;
}

File: 백준/Gold/1806. 부분합/부분합.py
import sys
input = sys.stdin.readline

n, s = map(int, input().split())
arr = list(map(int, input().split()))

st, ed = 0, 0
ans = 1e9
sum = arr[0]

while ed < n:
    if sum >= s: # 합이 넘으면 st 한 칸 이동
        ans = min(ans, ed - st + 1)

        sum -= arr[st]
        if st == ed:
            ed += 1
        st += 1
    else: # 합이 안 넘으면 ed 한 칸 이동
        ed += 1
        if ed < n:
            sum += arr[ed]

if ans == 1e9:
    ans = 0

print(ans)